src/Raven.Server/Documents/Replication/DocumentReplicationLoader.cs

```diff
@@ -728,9 +730,44 @@ public void Dispose()
             foreach (var outgoing in _outgoing)
                 ea.Execute(outgoing.Dispose);
 
+            _database.DocumentTombstoneCleaner?.Unsubscribe(this);
+
             ea.ThrowIfNeeded();
         }
 
+       
+        public Dictionary<string, long> GetLastProcessedDocumentTombstonesPerCollection()
```

Comment:
Might be just easier to change the tombstone deleter to just accept something that says "across all collections, my latest is X"